pkg/shell/base_index.js

```diff
@@ -496,7 +498,84 @@ function Index() {
     if (typeof self.navigate !== "function")
         throw Error("Index requires a prototype with a navigate function");
 
-    self.frames = new Frames(self);
+    /* Session timing out after inactivity */
+    let session_timer = null;
+    let session_final_timer = null;
+    let session_timeout = 0;
+    let final_countdown = 30000; // last 30 seconds
+    let title = "";
+
+    function startTimer() {
+        if (session_timeout > 0)
+            session_timer = window.setTimeout(sessionTimeout, session_timeout - final_countdown);
+    }
+
+    function sessionTimeout() {
+        /* When non-password login is in use, we don't want to logout used, as the user can just log
+         * back in. This can be recognized by checking `privileges` but this is set up
+         * asynchronously and it may happen that it is not yet set up in StartTimer().
+         * We can just check this out on first sessionTimeout() and if non-pwd login is used then
+         * just remove the timer and do not bother with it anymore.
+         */
+        if (self.privileges.is_locked() == true) {
```

Comment:
`== true` is entirely redundant, no?

I'd like @mvollmer to comment on this part. Reading pkg/shell/privileges.js I'm not sure if that's the right property to look at, especially with the sudo reorg that's starting now.

Reply:
The "locked" state seems to mean: "password was requested for sudo escalation but cockpit-ws doesn't have one".  This would be true whenever the user doesn't check "[ ] Reuse my password" on the login page.
So I think this is the wrong flag to use.
More generally, I think we can just log the user out even for SSO scenarios.

Reply:
But going back to the login page would just immediately log the user back in with SSO, no? If that is so, then the net effect is a confusing page reload; and if that's not so, a login page in an SSO case is a place from where there's no obvious path to get on, except for manually going back to the same URL? So I'm still fairly convinced that we should disable that if we don't have a password.